### PR TITLE
Add nunjucks configuration possibility

### LINKED ISSUE #16267: 🚀 Feature: Make nunjucks configureable, such as trimBlocks or lstripBlocks
### 🔖 Feature description

This feature would allow users to config nunjucks such as [trimBlocks or lstripBlocks](https://mozilla.github.io/nunjucks/api.html#configure).

### 🎤 Context

Right now, if I am defining a template file:

```
option 1
{% if values.option2NotEmpty %}
option 2
{% endif %}
option 3
```

when option 2 is empty, what I want is:

```
option 1
option 3
```

but not:

```
option 1

option 3
```

Nunjucks [trimBlocks](https://mozilla.github.io/nunjucks/api.html#configure) config can "_automatically remove trailing newlines from a block/tag_", but I did not find how to config this via backstage. From [SecureTemplater.ts](https://github.com/backstage/backstage/blob/v1.10.1/plugins/scaffolder-backend/src/lib/templating/SecureTemplater.ts#L32):

```
  const env = module.exports.configure({
    autoescape: false,
    tags: {
      variableStart: '\${{',
      variableEnd: '}}',
    },
  });
```

Backstage change nunjucks `autoescape` config from default `true`, but leave `trimBlocks` to default `false`, and no more config on the followed codes (`additionalTemplateGlobals` is for [another purpose](https://mozilla.github.io/nunjucks/api.html#addglobal)). I have hacked "node_modules/@backstage/plugin-scaffolder-backed/dist/index.cjs.js" which embedded SecureTemplater.ts:

```
  const env = module.exports.configure({
    autoescape: false,
    ......
    trimBlocks: true,
  });
```

then the result is what I want, so I am sure this config works.

### ✌️ Possible Implementation

There are some options:

* Like [make nunjucks' globals configureable](https://github.com/backstage/backstage/pull/13704), make nunjucks configureable.
* Set node env `NUNJUCKS_TRIM_BLOCKS=true` and change SecureTemplater.ts to use this, I am not sure that backstage can use node env var in SecureTemplater.ts?
* Like nunjucks [whitespace control](https://mozilla.github.io/nunjucks/templating.html#whitespace-control) (`{{-`), we need some signs to "trim blocks" at the block/tag scope. I think this is more flexible than global scope config, of course this is not backstage but nunjucks issue, I did not find this solution at [nunjucks](https://github.com/mozilla/nunjucks/issues/) or [jinja](https://jinja.palletsprojects.com/en/3.1.x/templates/#whitespace-control).

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by kissmikijr**
## Hey, I just made a Pull Request!

Fixes #16267
Add possibility to configure trimBlocks and lstripBlock nunjucks options with the `fetch:template` action
https://mozilla.github.io/nunjucks/api.html#configure

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by kissmikijr** [comment]
@benjdlambert Is this something that you were referring to? If this looks ok I'll add some tests for it

File: plugins/scaffolder-backend/src/lib/templating/SecureTemplater.ts [hunk 1]
```diff
@@ -109,6 +112,7 @@ export interface SecureTemplaterOptions {
   templateFilters?: Record<string, TemplateFilter>;
   /* Extra user-provided nunjucks globals */
   templateGlobals?: Record<string, TemplateGlobal>;
+  nunjucksConfigs?: NunjucksConfigs;
```
Comment (benjdlambert) [inline plugins/scaffolder-backend/src/lib/templating/SecureTemplater.ts:115]: can we just inline the type here rather than exporting a new one?

Comment (benjdlambert) [reply plugins/scaffolder-backend/src/lib/templating/SecureTemplater.ts:115]: Don't think we need to export the type actually in the first place as the `SecureTemplater` isn't exported.

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Nice - let's go with this :tada:

**Pull request merged**